### PR TITLE
MdePkg BootManagerPolicy.h: Define GUID for connecting storage devices.

### PR DESCRIPTION
**Pull request by apop5**
# Description
Some platforms require connecting storage media while booting to network, or require enumerating storage protocols that were not initially enumerated during BDS. This change adds a GUID to allow implementation of boot manager's ConnectDeviceClass to connect storage media.

- [ ] Breaking change?
- [ ] Impacts security?
- [ ] Includes tests?

## How This Was Tested
Platform implemented a custom BootMangerPolicy Driver which supported this guid and would connect the storage devices in the system.

## Integration Instructions
N/A

### CODE REVIEW

**Comment by mdkinney** [comment]
Is this change related to a UEFI Specification update?  Which version?  Please add references to the UEFI Spec version that applies to the protocol file header and the commit message and the PR summary.

File: MdePkg/Include/Protocol/BootManagerPolicy.h
Comment (abdattar) [inline MdePkg/Include/Protocol/BootManagerPolicy.h:37]: I think 'x' should be small case letter.

Comment (niruiyu) [inline MdePkg/Include/Protocol/BootManagerPolicy.h:37]: Has it been defined in Uefi Spec?

Comment (apop5) [reply MdePkg/Include/Protocol/BootManagerPolicy.h:37]: Yes, it was updated. Its in 2.11,  mantis item: 0002443

https://github.com/UEFI/UEFI_spec/commit/20971078eb7cb0ced138f56f9c7f27d0692d0a61

Comment (lgao4) [reply MdePkg/Include/Protocol/BootManagerPolicy.h:37]: This can be merged after UEFI 2.11 is published.

Comment (apop5) [reply MdePkg/Include/Protocol/BootManagerPolicy.h]: good catch. Fixed in latest push.

**Comment by apop5** [comment]
Not stale, waiting for release of specification.

**Comment by apop5** [comment]
Uefi 2.11 is published, are we ready to merge this?

**Review by lgao4 — APPROVED** [review]

**Comment by lgao4** [comment]
> Uefi 2.11 is published, are we ready to merge this?

Agree.

**Pull request merged**
